node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/QuestMetricsDb.java

```diff
@@ -196,8 +201,15 @@ public void gc() {
 
     @Override
     public void close() {
-        if (engine != null)
-            engine.close();
+        if (closed.get()) return;
+        closed.set(true);
```

Comment:
you need to use if (closed.getAndSet(true)) return to make it atomic.
If not there is a race betwene get() and set(true).